agent/replacecall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -345,44 +338,112 @@ private InvokeInstruction getReplacementInstruction(
     InvokeInstruction origInvocation = (InvokeInstruction) inst;
     MethodSignature origSig = MethodSignature.of(origInvocation, pool);
     MethodSignature newSig = replacementMap.get(origSig);
+
+    String super_class_name = cg.getSuperclassName();
+    String class_name = cg.getClassName();
+    String method_name = mg.getName();
+    String invoke_class = origSig.getClassname();
+    String invoke_method = origSig.getName();
+
+    //debug_transform.log("invoke_target: %s, invoke_class: %s, invoke_method: %s%n",
+    //    invoke_target, invoke_class, invoke_method);
+    //debug_transform.log("super: %s, host: %s.%s, target: %s%n", super_class_name, class_name, method_name, invoke_target);
+
     if (newSig == null) {
-      debug_transform.log(
-          "%s.%s: No replacement for %s%n", mg.getClassName(), mg.getName(), origSig.toString());
+      debug_transform.log("%s.%s: No replacement for %s%n", class_name, method_name, origSig);
       return null;
     }
 
+    // The code generated by the Java compiler for invoking a constructor
+    // follows this general pattern:
+    //
+    //      new    <{name of object}>
+    //      dup
+    //      {code to setup any arguments to the constructor}
+    //      invokespecial {name of object}.<init>
+    //
+    // The object created by the new instruction has not been initialized
+    // until ithe constructor completes.  If we simply replace the constructor
+    // call with a call to a replacement method, the JVM will fault on the
+    // uninitialized object reference.  So we must delete the new and dup
+    // instructions and have the replacement constructor supply a new object
+    // reference.
+    //
+    // The following code searches backwards from the invokespecial in the
+    // instruction list until it finds a new/dup pair and then deletes them.
```

Comment:
What if one of the arguments to the constructor is itself the result of `new MyOtherClass()`, as in
`new MyClass(new MyOtherClass())`?
Should it search only for new/dup pairs of the correct type?

Reply:
I included that case in my test and fixed CallReplaceTransformer to support it.